app/src/main/java/com/github/se/polyfit/viewmodel/dailyRecap/DailyRecapViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package com.github.se.polyfit.viewmodel.dailyRecap
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.se.polyfit.data.repository.MealRepository
+import com.github.se.polyfit.model.meal.Meal
+import dagger.hilt.android.lifecycle.HiltViewModel
+import java.time.LocalDate
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+@HiltViewModel
+class DailyRecapViewModel @Inject constructor(private val mealRepository: MealRepository) :
+    ViewModel() {
+  private val _meals: MutableStateFlow<List<Meal>> = MutableStateFlow(mutableListOf())
+  val meals: StateFlow<List<Meal>> = _meals
+
+  private val _isFetching: MutableStateFlow<Boolean> = MutableStateFlow(false)
+  val isFetching: StateFlow<Boolean> = _isFetching
```

Comment:
StateFlow also reacts to events that update the values (i.e. if you switch to an old date that has meals, when we set the meal value it will refresh the screen. Further, it provides default values versus having potential null values, which is why I went with it. Do you think that LiveData will provide any additional benefits on this purpose?